### DISCUSSION TITLE
Works with Notifications and without Redis

### DISCUSSION THREAD

**Discussion by beard7** [spatie/laravel-rate-limited-job-middleware#45] [Q&A] [answered]
Hi,

Just a quick question before I dive in...

Will this work with Notifications or basically any class that implements `ShouldQueue`

Is Redis required? I'm just using the database queue driver.

Thanks

Sorry... just realised this should have been posted in Discussions

**Comment by riasvdv** [comment, marked as answer]
> Will this work with Notifications or basically any class that implements ShouldQueue

As far as I know it should, but it's worth testing yourself as I haven't done this

As for the Redis requirement, you can pass `false` to the constructor and it will use your cache driver instead: https://github.com/spatie/laravel-rate-limited-job-middleware/blob/main/src/RateLimited.php#L29
